### PR TITLE
ISSUE 5134 - Update Intercom createContact attributes

### LINKED ISSUE #5134: Update Intercom contact creation
### Description
Update Intercom contact creation to include default attributes e.g. `unsubscribed_from_emails`

### Goal
- Should make use of fields Asite team also uses

### Tasks
- [x] `unsubscribed_from_emails`
- [x] `signed_up_at`
- [x] Check with team what to do with `company_entered`
  - [x] Team uses custom attribute `company` so it is preferred to copy value into this custom attribute

### PR DESCRIPTION
**Pull request by Charence**
This fixes #5134 

#### Description
- Additional attributes for Intercom contact

#### Test cases
- Should appear in Intercom

### CODE REVIEW

File: backend/src/v4/models/intercom.js [hunk 1]
```diff
@@ -46,14 +46,15 @@ Intercom.setIntercomHash = (userProfile) => {
 		.digest("hex");
 };
 
-Intercom.createContact = async (external_id, name, email, subscribed, company /* , job_title, phone_number, industry, found_us*/) => {
+Intercom.createContact = async (external_id, name, email, subscribed, company, createdAt /* , job_title, phone_number, industry, found_us*/) => {
```
Comment (carmenfan) [inline backend/src/v4/models/intercom.js:49]: Can we remove the commented out stuff?

File: backend/src/v4/models/intercom.js [hunk 2]
```diff
@@ -46,14 +46,15 @@ Intercom.setIntercomHash = (userProfile) => {
...
 	if (!accessToken) {
 		return;
 	}
 
 	const custom_attributes = {subscribed /* , job_title, industry, found_us*/ };
 	if (company) {
 		custom_attributes.company_entered = company;
+		custom_attributes.company = company;
```
Comment (carmenfan) [inline backend/src/v4/models/intercom.js:57]: is there much point in doing this? Shouldn't we just have one field?

File: backend/src/v4/models/user.js [hunk 3]
```diff
@@ -479,14 +479,14 @@ User.verify = async function (username, token, options) {
 	}
 
 	try {
-		const { customData: {firstName, lastName, email, billing, mailListOptOut /* , extras*/ } } = user;
+		const { customData: {firstName, lastName, email, billing, mailListOptOut, createdAt /* , extras*/ } } = user;
```
Comment (carmenfan) [inline backend/src/v4/models/user.js:482]: remove commented out stuff

File: backend/src/v4/models/user.js [hunk 4]
```diff
@@ -479,14 +479,14 @@ User.verify = async function (username, token, options) {
...
 		// const { jobTitle, phoneNumber, industry, howDidYouFindUs } = extras;
 
 		const subscribed = !mailListOptOut;
 		const company = get(billing, "billingInfo.company");
 
 		await Intercom.createContact(username, formatPronouns(firstName + " " + lastName), email,
-			subscribed, company /* , jobTitle, phoneNumber, industry, howDidYouFindUs */);
+			subscribed, company, createdAt /* , jobTitle, phoneNumber, industry, howDidYouFindUs */);
```
Comment (carmenfan) [inline backend/src/v4/models/user.js:489]: same here

File: backend/tests/v5/unit/processors/users.test.js [hunk 5]
```diff
@@ -225,6 +225,11 @@ const testGenerateResetPasswordToken = () => {
 
 const testSignUp = () => {
 	describe('Sign up a user', () => {
+		beforeAll(() => {
```
Comment (carmenfan) [inline backend/tests/v5/unit/processors/users.test.js:228]: do you need this at before all if only one test needed it?

Could we put it inside the test instead?

**Review by carmenfan — CHANGES_REQUESTED** [review]

File: backend/src/v4/models/intercom.js (hunk 2 above)
Comment (Charence) [reply backend/src/v4/models/intercom.js:56]: `company_entered` removed.

**Review by carmenfan — APPROVED** [review]

**Pull request merged**